@here/harp-mapview/lib/PickListener.ts

```diff
@@ -133,9 +148,22 @@ export class PickListener {
     }
 
     private sortResults(): void {
-        if (!this.m_sorted) {
-            this.m_results.sort(defaultSort);
-            this.m_sorted = true;
+        if (this.m_sorted) {
+            return;
         }
+
+        // HARP-14531: group and sort zero-distance results separately and merge them back
+        // to ensure that screen-space objects (e.g. labels) are returned first,
+        // as they are currently rendered on top.
+        const zeroDistanceGroup: PickResult[] = [];
+        const nonZeroDistanceGroup: PickResult[] = [];
+
+        this.m_results.forEach(result =>
+            (result.distance === 0 ? zeroDistanceGroup : nonZeroDistanceGroup).push(result)
+        );
+        this.m_results = zeroDistanceGroup
```

Comment:
Why not sort `this.m_results` first, then you don't need to sort the `zeroDistanceGroup` and the `nonZeroDistanceGroup`, because they are already sorted?

Reply:
That's true! Updated, thanks!